lti_consumer/tests/unit/test_lti.py

```diff
@@ -0,0 +1,287 @@
+        """
+        Test `get_signed_lti_parameters` returns the correct dict
+        """
+        expected_lti_parameters = {
+            u'user_id': self.lti_consumer.xblock.user_id,
+            u'oauth_callback': u'about:blank',
+            u'launch_presentation_return_url': '',
+            u'lti_message_type': u'basic-lti-launch-request',
+            u'lti_version': 'LTI-1p0',
+            u'roles': self.lti_consumer.xblock.role,
+            u'resource_link_id': self.lti_consumer.xblock.resource_link_id,
+            u'lis_result_sourcedid': self.lti_consumer.xblock.lis_result_sourcedid,
+            u'context_id': self.lti_consumer.xblock.context_id,
+            u'lis_outcome_service_url': self.lti_consumer.xblock.outcome_service_url,
+            'lis_person_sourcedid': 'edx',
+            'lis_person_contact_email_primary': 'edx@example.com',
+            u'custom_param_1': 'custom1',
+            u'custom_param_2': 'custom2',
+            u'oauth_nonce': '80966668944732164491378916897',
+            'oauth_timestamp': '1378916897',
+            'oauth_version': '1.0',
+            'oauth_signature_method': 'HMAC-SHA1',
+            'oauth_consumer_key': 't',
+            'oauth_signature': u'frVp4JuvT1mVXlxktiAUjQ7/1cw='
+        }
+        self.lti_consumer.xblock.has_score = True
+        self.lti_consumer.xblock.ask_to_send_username = True
+        self.lti_consumer.xblock.ask_to_send_email = True
+
+        self.lti_consumer.xblock.runtime.get_real_user.return_value = Mock(email='edx@example.com', username='edx')
+        self.assertEqual(self.lti_consumer.get_signed_lti_parameters(), expected_lti_parameters)
+
+        self.lti_consumer.xblock.runtime.get_real_user.return_value = {}
+        del expected_lti_parameters['lis_person_sourcedid']
+        del expected_lti_parameters['lis_person_contact_email_primary']
+        self.assertEqual(self.lti_consumer.get_signed_lti_parameters(), expected_lti_parameters)
+
+
+class TestLti2Consumer(TestLtiConsumerXBlock):
+    """
+    Unit tests for Lti2Consumer
+    """
+
+    def setUp(self):
+        super(TestLti2Consumer, self).setUp()
+        self.lti_consumer = Lti2Consumer(self.xblock)
+
+    def test_get_result(self):
+        """
+        Test `get_result` returns valid json response
+        """
+        self.xblock.module_score = 0.9
+        self.xblock.score_comment = 'Great Job!'
+        response = dict(GET_RESULT_RESPONSE)
+        response.update({
+            "resultScore": self.xblock.module_score,
+            "comment": self.xblock.score_comment
+        })
+        self.assertEqual(self.lti_consumer.get_result(Mock()), response)
+
+        self.xblock.module_score = None
+        self.xblock.score_comment = ''
+        self.assertEqual(self.lti_consumer.get_result(Mock()), GET_RESULT_RESPONSE)
+
+    @patch('lti_consumer.lti_consumer.LtiConsumerXBlock.clear_user_module_score')
+    def test_delete_result(self, mock_clear):
```

Comment:
Is there a reason you can't actually clear data as part of this test? Like perhaps mock the underlying data store? It's great we assert that the delete is called, but I'd want to confirm it actually deletes...